packages/babel-helpers/src/helpers.js

```diff
@@ -1358,17 +1363,27 @@ helpers.decorate = helper("7.1.5")`
   ) {
     var proto = F.prototype;
 
-    ["method", "field"].forEach(function(kind) {
-      elements.forEach(function(element /*: ElementDescriptor */) {
-        var placement = element.placement;
-        if (
-          element.kind === kind &&
-          (placement === "static" || placement === "prototype")
-        ) {
-          var receiver = placement === "static" ? F : proto;
+    elements.forEach(function(element /*: ElementDescriptor */) {
+      var placement = element.placement;
+      if (
+        element.kind === "method" &&
+        (placement === "static" || placement === "prototype")
+      ) {
+        var receiver = placement === "static" ? F : proto;
+        _defineClassElement(receiver, element);
+      }
+    });
+
+    elements.forEach(function(element /*: ElementDescriptor */) {
```

Comment:
Why two loops instead of combining the two?

Reply:
Becose they differ both in the if condition and in the behavior inside the if.

Reply:
I just noticed that I misinterpreted your question 🙃

The proposal says that all methods must be defined before the fields/initializers. If I used a single loop, that order wouldn't be maintained.